File: 3304-find-the-k-th-character-in-string-game-i/3304-find-the-k-th-character-in-string-game-i.cpp
class Solution {
public:
    char kthCharacter(int k) {
        string s = "a";
        while (s.size() < k) {
            string t = s;
            int n = t.size();
            for (int i=0; i<n; i++) {
                int x = t[i] - 'a';
                x = (x+1)%26;
                t[i] = 'a' + x;
            }
            s += t;            
        }
        return s[k-1];
    }
};

File: 3304-find-the-k-th-character-in-string-game-i/3304-find-the-k-th-character-in-string-game-i.py
class Solution:
    def kthCharacter(self, k: int) -> str:
        s = "a"
        while len(s) < k:
            n = len(s) 
            for i in range(n):
                x = ord(s[i]) - ord('a')
                x = (x+1) % 26 
                s += chr(x + ord('a'))
        return s[k-1]